### ISSUE TITLE
Command palette always scrolling to the top after click on a command

### ISSUE THREAD

**Issue by garu57** [jupyterlab/jupyterlab#4499]
Anaconda jupyterlab   0.32.0 on Chrome, Python 3.6.5 on win 10.

Perhaps i missed something about  `Command palette` usage but i  didn't find anything in doc to understand if i'm doing something wrong.

I tried to use it to learn available commands but each time i click on any of them, it immediately scrolls to the top, making repeatedly using the same command or commands of the same group a real hassle.

Also using the search is difficult; if i search for a command and click on it, the search box is cleared and the palette is scrolled again to the top. If i select a command group, for example `notebook cell operations` and then click on a command, again the search box is cleared and the palette is scrolled to the top.

I was thinking of moving cells, repeatedly clicking on `Move Cells Up/Down`, but with this behavior the palette is practically unusable.
To be usable it should stay put where it was scrolled or positioned by search.
Also, it would be useful adding a clear button for the search box. If, for example, i select a command group, as i told before, then i change idea and want to get back to the whole list, i have to clear the search box with the keyboard.

**Comment by sccolbert** [comment]
Duplicate of #3923 wrt scrolling after click with nothing in the search bar.

Clearing the search bar after command execution is intentional. We previously had the behavior that the search text was simply selected after execution, so the next character typed would clear the text. But the consensus was that that was less useful than clearing all the text.

**Comment by sccolbert** [comment]
cc @ellisonbg @jasongrout

**Comment by jasongrout** [comment]
Traditionally in coding environments, a command palette is something that is popped up, searched, command executed, and then goes away. That interaction scenario works in our design choices. However, our command palette by nature is also more persistent, being in a side panel, which lends itself to the scenarios you describe, which is a totally different interaction behavior. Perhaps we should have a setting to switch between the two behaviors? Or perhaps we should have the side panel command palette persist the search, but a popup command palette for the 'search/execute/close' scenario?

**Comment by garu57** [comment]
>  However, our command palette by nature is also more persistent, being in a side panel, which lends itself to the scenarios you describe, which is a totally different interaction behavior. Perhaps we should have a setting to switch between the two behaviors?

@jasongrout yes please! I find that the palette contains two basic types of commands, those that are apt to one shot usage, and those, like the ones contained in `notebook cell operations` and `notebook operations` groups, that can be used to interact with notebooks.

For instance the nice effect of putting the selected command group in search box, in my 4k monitor, is that clicking the  `notebook operations` group i see that it and the other, `notebook cell operations` are in the search result, perfectly fitting in the vertical and giving me at a glance all the needed commands to interact with notebooks (i'm too lazy to remember all the shortcuts).

But to be used in this way i think we need two things:

- the search result or the scrolling position must be made persistent
- we need a button to clear the search box

> Or perhaps we should have the side panel command palette persist the search, but a popup command palette for the 'search/execute/close' scenario?

I cannot judge which solution is the best, obviously it's a personal opinion, but i find that popup command palettes suffer from a basic limitation when learning, you must know in advance what you are looking for and you really don't.
When i saw you implemented the command palette as a plain menu i shouted a big hurrah!

**Comment by chrisjsewell** [comment]
Definitiely +1 for this 😃

> Traditionally in coding environments, a command palette is something that is popped up, searched, command executed, and then goes away.

I'd argue it's not actually a command palette though. In the terminology of VS Code (which Jupyter lab very much seems to converging towards) it is a 'View Container' (see [here](https://code.visualstudio.com/api/extension-capabilities/extending-workbench)), with the command palette being a separate entity (see [here](https://code.visualstudio.com/docs/getstarted/userinterface#_command-palette))

Basically if you can just make Jupyter lab like a web based version of VS Code, that would be great lol!

**Comment by ellisonbg** [comment]
I am very much in favor of pulling the currently JupyterLab command palette out of the left side panel and making a model interaction that hovers over the workspace in a manner that is similar to VS Code. I think there is some interaction design to do on how it all works, but I think that will be a more effective way of exposing keyboard driven actions.
